src/main/java/org/folio/service/impl/AllowedServicePointsServiceImpl.java

```diff
@@ -48,19 +49,32 @@ public AllowedServicePointsResponse getAllowedServicePoints(AllowedServicePoints
   }
 
   public AllowedServicePointsResponse getForCreate(AllowedServicePointsRequest request) {
-    String instanceId = request.getInstanceId();
+    String instanceId = "";
     String patronGroupId = userService.find(request.getRequesterId()).getPatronGroup();
     log.info("getForCreate:: patronGroupId={}", patronGroupId);
-
-    var searchInstancesResponse = searchClient.searchInstance(instanceId);
-    // TODO: make call in parallel
-    boolean availableForRequesting = searchInstancesResponse.getInstances().stream()
-      .map(Instance::getItems)
-      .flatMap(Collection::stream)
-      .map(Item::getTenantId)
-      .filter(Objects::nonNull)
-      .distinct()
-      .anyMatch(tenantId -> checkAvailability(request, patronGroupId, tenantId));
+    boolean availableForRequesting = false;
+    if (request.isForTitleLevelRequest()) {
+      instanceId = request.getInstanceId();
+      var searchInstancesResponse = searchClient.searchInstance(instanceId);
+      // TODO: make call in parallel
+      availableForRequesting = searchInstancesResponse.getInstances().stream()
+        .map(Instance::getItems)
+        .flatMap(Collection::stream)
+        .map(Item::getTenantId)
+        .filter(Objects::nonNull)
+        .distinct()
+        .anyMatch(tenantId -> checkAvailability(request, patronGroupId, tenantId,
+          request.getInstanceId()));
+    }
+    if (request.isForItemLevelRequest()) {
+      String itemId = request.getItemId();
+      var searchItemResponse = searchClient.searchItem(itemId);
+      if (StringUtils.isNotEmpty(searchItemResponse.getTenantId())) {
+        instanceId = searchItemResponse.getInstanceId();
+        availableForRequesting = checkAvailability(request, patronGroupId,
+          searchItemResponse.getTenantId(), instanceId);
+      }
+    }
```

Comment:
Since you already have an `itemId`, you can call mod-circulation's allowed-service-point with `itemId` instead of `instanceId` and without `useStubItem` flag.